runtime/bcutil/cfreader.c

```diff
@@ -1783,11 +1798,15 @@ checkMethods(J9PortLibrary* portLib, J9CfrClassFile* classfile, U_8* segment, U_
 		} 
 
 		if (nameIndexOK && utf8Equal(&classfile->constantPool[method->nameIndex], "<init>", 6)) {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES) && defined(J9VM_OPT_NEW_FACTORY_METHOD)
```

Comment:
The `<new>` unnamed factory is for Valhalla only, better to rename `J9VM_OPT_NEW_FACTORY_METHOD` to `J9VM_OPT_VALHALLA_NEW_FACTORY_METHOD` and always put it inside `#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)`. i.e.
```
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
#if defined(J9VM_OPT_NEW_FACTORY_METHOD)
/* code here */
#endif /* defined(J9VM_OPT_NEW_FACTORY_METHOD) */
#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```